pallets/identity/src/lib.rs

```diff
@@ -255,6 +261,31 @@ decl_module! {
             use polymesh_primitives::{identity_claim::IdentityClaimOld, migrate::migrate_map};
             migrate_map::<IdentityClaimOld>(b"Identity", b"Claims");
 
+            // Covert old scopes to new scopes
```

Comment:
Let's extract this logic to `polymesh_primitives::migrate` and abstract away concrete details, including:
- The key types, new and old (4 type params)
- The hasher types, for each key, new and old (4 type params)
- A type parameter for the value type being passed through.
- A mapping `F: FnMut(K1, K2) -> (K1New, K2New)` closure.
- The storage `module::item`.

It's a lot parameters, much of it repetitive (although you can reduce it by defining another function that avoids changing the hashers, which I think is the common case, cutting back 4 parameters), but I think its preferable to open-coding.

Reply:
I'll capture this is a separate task in the backlog. Something we can also push upstream.